### PR TITLE
5560: [TEST]: Add automatic cleanup in functional test

### PR DESCRIPTION
**Pull request by pkazlenka**
Implements #5560

* Added new mechanism to automatically clean up after tests (no need to fulfill 'cleanup' section in spock
* Applied new mechanism to regular flow deletion

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CleanupVerifierListener.groovy [hunk 1]
```diff
@@ -93,7 +94,7 @@ class CleanupVerifierListener extends AbstractSpringListener {
         regionVerifications.verify()
         withPool {
             database.getIsls(topology.isls).eachParallel {
-                assert it.timeUnstable == null
+                //assert it.timeUnstable == null
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CleanupVerifierListener.groovy:97]: typo

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy [hunk 2]
```diff
@@ -131,11 +137,21 @@ class FlowHelper {
      */
     FlowPayload addFlow(FlowPayload flow) {
         log.debug("Adding flow '${flow.id}'")
+        cleanupManager.addAction({flowHelperV2.safeDeleteFlow(flow.getId())}, CleanupAfter.TEST)
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy:140]: It seems like 'flowHelperV2.safeDeleteFlow' is not a safe way of flow deletion, and we can get an error during this action(maybe we need to skip waiting for a history deletion event, and just wait for flow absence). As a result, the rest of the auto-cleanup closures will be skipped.

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/model/cleanup/CleanupManager.groovy [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * This class handles disposable objects (flows, switch settings, turned off ISLs, etc) and deletes them or restores
+ * their state after test or test class run (when called in AutomaticCleanupListener)
+ */
+@Component
+@Scope("specThread")
+class CleanupManager {
+    @Value("#{T(java.util.Collections).emptyMap()}")
+    private HashMap<CleanupAfter, List<Closure>> cleanupActions
+    @Value("#{T(java.util.Collections).emptyMap()}")
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/model/cleanup/CleanupManager.groovy:21]: typo: unnecessary line

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy [hunk 4]
```diff
@@ -76,7 +76,6 @@ class AutoRerouteSpec extends HealthCheckSpecification {
         }
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy:76]: Potential flaky place: 
It seems like the logic of clean-up has been changed as we switch on the port before link deletion. That can cause the auto reroute of the flow. It is a potential problem during deletion as this flow can be in an 'In progress' state. Maybe for these cases, we need to wait for the reroute completion.

**Review by yuliiamir — APPROVED** [review]

**Review by dmitrii-beliakov — APPROVED** [review]

**Pull request merged**